### PR TITLE
Upstreaming ports and fixes from DevilutionX

### PR DESCRIPTION
**Pull request by AJenbo**
### Ports
Amiga
Linux BigEndian
Nintendo Switch
Microsoft Xbox

### Fixes
One typo
Missing macOS headers
Incorrect default for hFile

### CODE REVIEW

File: src/SFileFindFile.cpp [hunk 1]
```diff
@@ -212,6 +216,7 @@ static bool DoMPQSearch_FileEntry(
     TMPQHash * pHashEntry,
     TFileEntry * pFileEntry)
 {
+#ifndef _XBOX
```
Comment (ladislav-zezula) [inline src/SFileFindFile.cpp:219]: Why is XBOX platform so special here, in regards of patch archives?

File: src/SCompression.cpp [hunk 2]
```diff
@@ -267,9 +267,11 @@ static void Compress_PKLIB(void * pvOutBuffer, int * pcbOutBuffer, void * pvInBu
         else
             dict_size = CMP_IMPLODE_DICT_SIZE3;
 
+#ifndef _XBOX
```
Comment (ladislav-zezula) [inline src/SCompression.cpp:270]: Implode not implemented in XBOX platform?

File: src/FileStream.cpp [hunk 3]
```diff
@@ -560,7 +565,12 @@ static bool BaseMap_Open(TFileStream * pStream, LPCTSTR szFileName, DWORD dwStre
         // Get the file size
         if(fstat64(handle, &fileinfo) != -1)
         {
+#if !defined(PLATFORM_AMIGA)
```
Comment (ladislav-zezula) [inline src/FileStream.cpp:568]: Please, don't replace memory mapping with memory allocation. It's OK to just leave it unsupported if the target platform does not support memory mapped files.

File: src/StormPort.h [hunk 4]
```diff
@@ -50,10 +50,12 @@
   #include <stdio.h>
 
   // Suppress definitions of `min` and `max` macros by <windows.h>:
+#ifndef _XBOX
```
[the remark this answers is not in the document]
Comment (AJenbo) [reply src/StormPort.h:53]: The Xbox platform does indeed derive from Windows, I think I will remove it from this PR for now as it is the least mature of the ports. I'll take your suggestions up with the developer that provided it.

File: src/FileStream.cpp (hunk 3 above)
Comment (AJenbo) [reply src/FileStream.cpp:568]: are you referring to the switch platform?

Comment (ladislav-zezula) [reply src/FileStream.cpp:568]: There is
```
#if !defined(PLATFORM_AMIGA)
#if defined(PLATFORM_SWITCH)
```
If memory mapping is supported under a platform, leave `mmap` there. If not, just leave unimplemented.

Comment (AJenbo) [reply src/FileStream.cpp:568]: are both wrong?

Comment (ladislav-zezula) [reply src/FileStream.cpp:568]: Yes. `mmap` is a function that maps file data to a memory. You get a pointer to memory that contains plain file data. `malloc` is not a replacement of it, by no means.

**Comment by AJenbo** [comment]
sorry i never gotaroundd to update this

**Pull request closed without merging**